### ISSUE TITLE
NullPointerException when retrieving body

### ISSUE THREAD

**Issue by cmaeder** [Sable/soot#809]
I've got a few apks that fail with: 

```
java.lang.NullPointerException
	at soot.jimple.toolkits.typing.fast.UseChecker.caseAssignStmt(UseChecker.java:362)
	at soot.jimple.internal.JAssignStmt.apply(JAssignStmt.java:238)
	at soot.jimple.toolkits.typing.fast.UseChecker.check(UseChecker.java:125)
	at soot.jimple.toolkits.typing.fast.TypeResolver.insertCasts(TypeResolver.java:426)
	at soot.jimple.toolkits.typing.fast.TypeResolver.minCasts(TypeResolver.java:437)
	at soot.jimple.toolkits.typing.fast.TypeResolver.inferTypes(TypeResolver.java:158)
	at soot.jimple.toolkits.typing.TypeAssigner.internalTransform(TypeAssigner.java:125)
	at soot.BodyTransformer.transform(BodyTransformer.java:59)
	at soot.BodyTransformer.transform(BodyTransformer.java:64)
	at soot.dexpler.DexBody.jimplify(DexBody.java:650)
	at soot.dexpler.DexMethod$1.getBody(DexMethod.java:146)
	at soot.SootMethod.getBodyFromMethodSource(SootMethod.java:126)
	at soot.SootMethod.retrieveActiveBody(SootMethod.java:385)
```

Is there something that can be done when the type cannot be reconstructed?

**Comment by jpstotz** [comment]
I also ran into this problem. The code in `UseChecker.caseAssignStmt` simply does not cover the case that `et` (estimated type?) can be null as none of the heuristics matches. 

Usually I would assume that in such a case there should be a fall-back may to `java.lang.Object`. However I don't understand the code completely to perform such a change.

**Comment by mbenz89** [comment]
Fixed in #877
